### DISCUSSION TITLE
Incoming connection signal/alert via Basic Interface?

### DISCUSSION THREAD

**Discussion by JerichoJack** [Povstalec/StargateJourney#55] [Q&A] [answered]
Long story short, i am wanting to get an incoming connection "alert" in my CC program... but i can't tell what value to look for....

Building an ATM9 server for the kiddo (to learn some programming via CC) and would like it if in my base i could get an alert if my Stargate (i took over Lantea) is activated via incoming connection.... I have attempted to use 2 interfaces, and essentially wrote the code that IF activity on interface 2 AND i did NOT start a dial, alarm... but nothing.... Any ideas would be greatly appreciated.

**Comment by Povstalec** [comment, marked as answer]
It would probably be good to use ComputerCraft events for that:
https://github.com/Povstalec/StargateJourney/wiki/Interfaces#events

You can monitor whenever a chevron is engaged, an incoming/outgoing wormhole starts forming and when entities enter/exit the Stargate

You can use this as a base, it's code that listens to ComputerCraft events produced by the Stargate. I made it the first time I added the events, so some of the variables it outputs may be in different order or may be nil on the Basic Interface, but the general principle hasn't changed for any of the events:

```
interface = peripheral.find("advanced_crystal_interface")

local address = {}



function chevronMonitoring()
	  
    while true
    do
        local event, chevron, incoming, symbol = os.pullEvent("stargate_chevron_engaged")
        print("Chevron ",chevron,"incoming ",incoming," symbol ",symbol)
        sleep(0)
    end
	
end

function outgoingWormholeMonitoring()
	  
    while true
    do
        local event, address = os.pullEvent("stargate_outgoing_wormhole")
        
		if #address > 0 then
		
			for symbol = 1,#address,1
			do
				print("Symbol ",symbol," = ", address[symbol])
			end
		
		end
		
        sleep(0)
    end
	
end

function incomingWormholeMonitoring()
	  
    while true
    do
        local event, address = os.pullEvent("stargate_incoming_wormhole")
        
		if #address > 0 then
		
			for symbol = 1,#address,1
			do
				print("Symbol ",symbol," = ", address[symbol])
			end
		
		end
		
        sleep(0)
    end
	
end

function deconstructionMonitoring()
	  
    while true
    do
        local event, entityType, name, uuid, disintegrated = os.pullEvent("stargate_deconstructing_entity")
        print("Type ",entityType," name ",name, " uuid ",uuid, " disintegrated ",disintegrated)
        sleep(0)
    end
	
end

function reconstructionMonitoring()
	  
    while true
    do
        local event, entityType, name, uuid = os.pullEvent("stargate_reconstructing_entity")
        print("Type ",entityType," name ",name, " uuid ",uuid)
        sleep(0)
    end
	
end

parallel.waitForAll(chevronMonitoring, outgoingWormholeMonitoring, incomingWormholeMonitoring, deconstructionMonitoring, reconstructionMonitoring)

```

**Reply by JerichoJack** [reply]
That is perfect! Exactly what I'm looking for! I'm already using CC, so this is perfect!

Thank you, this is a really cool mod, and the kiddo is going to have a blast learning about code structure with CC.

**Reply by Povstalec** [reply]
Have fun
